docs/examples/kops-tests-private-net-bastion-host.md

```diff
@@ -0,0 +1,379 @@
+```
+
+Some things to note from here:
+
+- "NAME" will be an environment variable that we'll use from now in order to refer to our cluster name. For this practical exercise, our cluster name is "privatekopscluster.k8s.local".
+- Because we'll use gossip DNS instead of a valid DNS domain on AWS ROUTE53 service, our cluster name need to include the string **".k8s.local"** at the end (this is covered on our AWS tutorials). You can see more about this on our [Getting Started Doc.](https://github.com/kubernetes/kops/blob/master/docs/aws.md)
+
+
+## KOPS PRIVATE CLUSTER CREATION:
+
+Let's first create our cluster ensuring a multi-master setup with 3 masters in a multi-az setup, two worker nodes also in a multi-az setup, and using both private networking and a bastion server:
+
+```bash
+kops create cluster \
+--cloud=aws \
+--master-zones=us-east-1a,us-east-1b,us-east-1c \
+--zones=us-east-1a,us-east-1b,us-east-1c \
+--node-count=2 \
+--topology private \
+--networking kopeio-vxlan \
+--node-size=t2.micro \
+--master-size=t2.micro \
+${NAME}
+```
+
+A few things to note here:
+
+- The environment variable ${NAME} was previously exported with our cluster name: privatekopscluster.k8s.local.
+- "--cloud=aws": As kops grows and begin to support more clouds, we need to tell the command to use the specific cloud we want for our deployment. In this case: amazon web services (aws).
+- For true HA at the master level, we need to pick a region with at least 3 availability zones. For this practical exercise, we are using "us-east-1" AWS region which contains 5 availability zones (az's for short): us-east-1a, us-east-1b, us-east-1c, us-east-1d and us-east-1e.
+- The "--master-zones=us-east-1a,us-east-1b,us-east-1c" KOPS argument will actually enforce that we want 3 masters here. "--node-count=2" only applies to the worker nodes (not the masters).
+- The "--topology private" argument will ensure that all our instances will have private IP's and no public IP's from amazon.
+- We are including the arguments "--node-size" and "master-size" to specify the "instance types" for both our masters and worker nodes.
+- And finally, the "--networking kopeio-vxlan" argument. With the private networking model, we need to tell kops which networking subsystem to use. More information about kops supported networking models can be obtained from the [KOPS Kubernetes Networking Documentation](https://github.com/kubernetes/kops/blob/master/docs/networking.md). For this exercise we'll use "kopeio-vxlan" (or "kopeio" for short).
+
+**NOTE**: You can add the "--bastion" argument here if you are not using "gossip dns" and create the bastion from start, but if you are using "gossip-dns" this will make this cluster to fail (this is a bug we are correcting now). For the moment don't use "--bastion" when using gossip DNS. We'll show you how to get around this by first creating the private cluster, then creation the bastion instance group once the cluster is running.
```

Comment:
This is a bug, can you reference the issue?

Reply:
Ok. Let me find the issue, but if you have the number and link... better :D